os/kernel/binary_manager/binary_manager.c

```diff
@@ -110,6 +112,27 @@ void binary_manager_register_partition(int part_num, int part_type, char *name,
 	}
 }
 
+/* Verify that loaded binary is running and update state */
+int binary_manager_update_state(int pid, int state)
+{
+	int bin_idx;
+
+	if (pid < 0 || state < 0 || state >= BINARY_STATE_MAX) {
+		dbg("Invalid parameter: pid %d, state %d\n", pid, state);
+		return BINMGR_INVALID_PARAM;
+	}
+
+	for (bin_idx = 0; bin_idx < binary_manager_get_binary_count() + 1; bin_idx++) {
```

Comment:
This is not good habit, return type of function can be changed, so you had better avoid use function as a break condition in loop.
Also function will be called for every loop.
If it is intended code, ignore my comment, otherwise please apply below
int count = binary_manager_get_binary_count();
for (idx = 0; idx < count; idx++)